xds/internal/resolver/watch_service.go

```diff
@@ -110,6 +110,25 @@ func (w *serviceUpdateWatcher) handleLDSResp(update xdsclient.ListenerUpdate, er
 		httpFilterConfig:  update.HTTPFilters,
 	}
 
+	if update.RouteConfigName == "" {
+		// RDS name from update is empty string, the LDS resp received has an
+		// inline RDS resource.
+
+		// If there was an RDS watch, cancel it.
+		if w.rdsName != "" {
+			w.rdsName = ""
+			if w.rdsCancel != nil {
+				w.rdsCancel()
+			}
+			w.rdsCancel = nil
+		}
+
+		// Handle the inline RDS update as if it's from an RDS watch.
+		w.updateVirtualHostsFromRDS(update.InlineRouteConfig)
+		return
+	}
+
+	// RDS name from update is not an empty string, start RDS watch.
```

Comment:
This seems like it's documenting the `if` immediately below, but it isn't.

Reply:
Modified (a little bit).